src/proxy-ui-api/src/main/java/org/niis/xroad/restapi/service/TokenCertificateService.java

```diff
@@ -372,6 +382,83 @@ private ClientId getClientIdForSigningCert(String instanceIdentifier, X509Certif
         return certificateSubject;
     }
 
+    /**
+     * Send the authentication certificate registration request to central server
+     * @param hash certificate hash
+     * @param securityServerAddress IP address or DNS name of the security server
+     * @throws CertificateNotFoundException
+     * @throws GlobalConfService.GlobalConfOutdatedException
+     */
+    public void registerAuthCert(String hash, String securityServerAddress) throws CertificateNotFoundException,
```

Comment:
This should in principle check business logic rules and process registration requests only if cert is in a state that allows it. Parts like this in old implementation:

```
        :register_enabled => key.usage == KeyUsageInfo::AUTHENTICATION &&
          [CertificateInfo::STATUS_SAVED].include?(cert.status),
```

But this logic will be implemented by XRDDEV-778 and 806. If one or both of those issues merges before this, it would be good to apply their validation mechanism to this. 806 offers a bit simpler way to do it. Would be something like this

```
possibleActionsRuleEngine.requirePossibleCertificateAction(PossibleActionEnum.REGISTER,
            tokenInfo, keyInfo, certificateInfo);
```

and would require lookup for tokenInfo, keyInfo (there's a new single operation getter for both based on cert hash).

If 778 and/or 806 are not merged before this PR, this is just FYI

Reply:
Maybe this task should be merged only after 778 and 806 since this is a smaller one

Reply:
Will be adding this check next

Reply:
Added and tested